.github/workflows/ci_4337.yml

```diff
@@ -14,14 +14,41 @@ jobs:
           node-version: 18.x
           cache: npm
           cache-dependency-path: 4337/package-lock.json
-      - run: npm ci 
-      - run: npm run build
+      - run: npm ci
+      - run: npm run build && npm run build:ts
       - run: npm run coverage
       - name: Coveralls
         uses: coverallsapp/github-action@master
         with:
           path-to-lcov: 4337/coverage/lcov.info
           github-token: ${{ secrets.GITHUB_TOKEN }}
+  e2e:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./4337
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+          cache: npm
+          cache-dependency-path: 4337/package-lock.json
+      - run: npm ci
+      - run: |
+          docker compose up -d
+          # wait for containers to start up
+          SECONDS=0
```

Comment:
TIL https://unix.stackexchange.com/a/646605

Reply:
It doesn't seem like the timeout works, though. It took longer than 30 seconds for cold boot on my local machine, and according to the CI logs, it also takes more than 30 seconds.

Reply:
This timeout only starts after the `docker componse up -d` command ends (I reset the counter to 0 here).

Building the docker image usually is what takes the longest (~1minute), but it isn’t counted in the timeout.

Reply:
Ahh it's for the start only, sorry I misunderstood it